proto/topo.proto

```diff
@@ -41,6 +41,7 @@ enum Vendor {
   GOBGP = 8;
   NOKIA = 9;
   OPENCONFIG = 10;
```

Comment:
Unfortunately there is a bit of a strange pattern developing here where we have an openconfig vendor that is always assumed to be lemming

Generally the pattern so far has been Vendor should be the same across all models that a vendor provides, for example vendor CISCO, model XRD or model 8000e. There is a single node impl for each vendor that can handle the specific bring up based on the model field value

However in this case I believe ondatra uses vendor without model to determine if the device is an ATE or not so we probably cant follow the same pattern with the openconfig vendor

Probably makes most sense to consider openconfig not as a vendor but instead the vendor is Magna or Lemming. This current PR works towards that, optionally there can be a follow up from Daniel to deprecate the openconfig vendor and create a lemming vendor in the future. @DanG100 

What are your thoughts? @robshakir

Reply:
i dug a bit into ONDATRA, where it does look pretty baked that we assume that any model of vendor `KEYSIGHT` is an ATE. i.e., opposite to the "assume all of vendor `OPENCONFIG` is lemming" we say "any model of `KEYSIGHT` is an ATE".
 
we could change this, but I wonder whether there's value in doing this -- since today I'm not aware of any "vendor" other than `OPENCONFIG` that provides both ATEs and devices.

maybe we could, rather than adding `MAGNA` here, we could add `OPENCONFIG_ATE`, and then use model `MAGNA`. the `topo/node` implementation can then be the place where the model is used to know to spin up magna. a similar change can be made to `topo/lemming` to know of different `OPENCONFIG` DUT implementations.

Reply:
Discussed a bit with @bormanp today. Seems like there are 4 options here:
1) vendor is MAGNA and we switch OPENCONFIG vendor to LEMMING as a follow up
2) vendor is OPENCONFIG_ATE
3) vendor is OPENCONFIG and we change the kne solver to use a combo of model + vendor
4) vendor is OPENCONFIG and we add a ATE field to the topo proto to differentiate without needing to know the model

I would lean toward 3 if we can successfully modify the solver. If not I would be in favor of 2. Let's finalize this early next week after we have a bit more consideration. Sorry for the delay

Reply:
in summary please move this magna implementation to the lemming.go file, differentiate based on the model value (if model == magna, model == lemming etc.) similar to cisco's approach for 8000e vs XRD. finally add the default value for the ondatra-role label

I will make an internal change to the solver to use the ondatra-role label and we can merge this change and the ondatra PR after that gets synced out

Reply:
Apologies for the delay responding -- was OOO for a bit.

That sounds good -- I'll refactor this as you suggest.